shared/utility/math/geometry/Quad.h

```diff
@@ -28,13 +29,209 @@ namespace utility {
 namespace math {
     namespace geometry {
 
+        // *********************
+        // * ARMADILLO VERSION *
+        // *********************
```

Comment:
If we're trying to move away from armadillo, rather than reimplementing things like the Quad class, should we look into bringing in a computational geometry library?

Reply:
That is a much bigger overhaul than what we are planning right now. This armadillo version is only surviving until the behaviour modules are converted to Eigen